File: Backend/SmartSolarMicrogrid.API/DTOs/Auth/LoginRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SmartSolarMicrogrid.API.DTOs.Auth;

/// Represents the login information submitted by a user.
public class LoginRequest
{
    /// NIC or email address used to identify the account.
    [Required]
    public string Identifier { get; set; } = string.Empty;

    /// Plain-text password
    [Required]
    public string Password { get; set; } = string.Empty;
}

File: Android/app/src/main/java/com/smartsolar/microgrid/model/LoginRequest.kt
package com.smartsolar.microgrid.model

/**
 * Request model used when a user logs into the system.
 *
 * The identifier can be a NIC or email address.
 */
data class LoginRequest(

    // NIC or email entered by the user.
    val identifier: String,

    // Password entered by the user.
    val password: String
)